4_other_sources/src/get_norwest_data.R

```diff
@@ -0,0 +1,178 @@
+# function to retreive all of the daily filenames for download
+get_filenames <- function(url) {
+  
+  # get html from downloads website
+  html <- paste(readLines(url), collapse = '\n')
+  matched <- str_match_all(html,  "<a href=\"(.*?)\"")
+  matched2 <- unlist(matched)
+  
+  # find the daily observations
+
+  zips <- grep(pattern = '\\.zip', x = matched2, value = TRUE)
+  dailies <- unique(grep('Daily', zips, value = TRUE))
+  dailies <- dailies[!grepl('href', dailies)]
+  dailies <- grep('AllDays', dailies, value = TRUE)
+
+  # find the site points
+  points <- unique(grep('^downloads.*ObservedTempPoints', zips, value = TRUE))
+  
+  # observation data files
+  obs_files <- gsub('downloads/ObservedStreamTemperatureMaps/', '', dailies)
+  obs_tasks <- gsub('/NorWeST.*', '', obs_files)
+  obs_tasks <- gsub('\\d|_|-', '', obs_tasks)
+
+  obs_files <- data.frame(data_files = obs_files, tasks = obs_tasks)
+  
+  # site data files
+  site_files <- gsub('downloads/ObservedStreamTemperatureMaps/', '', points)
+  site_tasks <- gsub('/NorWeST.*', '', site_files)
+  site_tasks <- gsub('\\d|_|-', '', site_tasks)
+  
+  site_files <- data.frame(site_files = site_files, tasks = site_tasks)
+  
+  # create data frame for downloads
+  file_downloads <- left_join(obs_files, site_files, by = 'tasks')
+  
+  return(file_downloads)
+}
+
+do_data_file_tasks <- function(files, base_url, out_file, ...) {
+  task_name <- files$tasks
+  
+  download_temp_data <- create_task_step(
+    step_name = 'download_temp_data',
+    target_name = function(task_name, ...) {
+      sprintf('%s_temperature_data', task_name)
+    },
+    command = function(task_name, ...) {
+      sprintf("fetch_data_files(base_url = download_base_url, files = dl_filenames, region = I('%s'))", task_name)
+    }
+  )
+  
+  download_site_data <- create_task_step(
+    step_name = 'download_site_data',
+    target_name = function(task_name, ...) {
+      sprintf('%s_site_data', task_name)
+    },
+    command = function(task_name, ...) {
+      sprintf("fetch_site_files(base_url = download_base_url, files = dl_filenames, region = I('%s'))", task_name)
+    }
+  )
+  
+  task_plan <- create_task_plan(
+    task_names = task_name, 
+    task_steps = list(download_temp_data, download_site_data),
+    add_complete = FALSE,
+    final_steps = c('download_temp_data', 'download_site_data')
+  )
+  # Create the task remakefile
+  create_task_makefile(
+    task_plan = task_plan,
+    makefile = '4_norwest_datafile_tasks.yml',
+    include = 'remake.yml',
+    sources = c(...),
+    packages = c('dplyr', 'stringr', 'sf', 'readxl'),
+    tickquote_combinee_objects = TRUE,
+    finalize_funs = c('combine_temp_data', 'combine_site_data'),
+    final_targets = c('4_other_sources/out/norwest_raw_temp_data.feather.ind', '4_other_sources/out/norwest_raw_site_data.rds.ind'),
+    as_promises = TRUE)
```

Comment:
From your overview comment on the PR:

> One thing I noticed is that there are no build files for `4_other_sources/out/norwest_raw_temp_data.feather` or `4_other_sources/out/norwest_raw_site_data.rds`.

I suspect this `as_promises = TRUE` is why the final target .inds aren't getting written - with that set, you're creating '*.ind_promise' files, which wouldn't trigger the creation of build/status files because they don't have the '.ind' suffix. (https://github.com/USGS-R/scipiper/blob/master/R/create_task_makefile.R#L151)

I think that the additional targets I suggested in 4_other_sources.yml would solve this.